File: Mighty_Screen.pl
my $contrast = 255;   # Need to rewrite on update Firmware
my $transparanse = 0; # Need to rewrite on update Firmware

my $colors;
my $sign;

my ($r, $g, $b);
my $sum;

$colors = <>;
$sign = <>;
chomp $colors;

($r, $g, $b) = split /,/, $colors;
$sum = $r + $g + $b;
	
say $contrast; # 1
say $r;        # 2
say $sum / 255; # 3 	
say $transparanse; # 4 

# 5
if (($r > $g) && ($r > $b)) {
  say $r;
} else {
  if ($g > $b) {
    say $g;
  } else {
    say $b;
  }
}	

# 6
if (($r < $g) && ($r < $b)) {
  say $r;
} else {
  if ($g < $b) {
    say $g;
  } else {
    say $b;
  }
}	

say $g;      # 7
print $sign; # 8

# by Tovarisch Trunaev, 2025, Russia, Uray
{ Om->Delf(); }

File: Mighty_Screen.pas
Program Mighty_Screen;
uses
  SysUtils;
Const
	Contrast = 255;   { Need to rewrite on update Firmware }
	Transparanse = 0; { Need to rewrite on update Firmware }
Var 
    colors: string;
    sign: string;
    
    rs, rg, rb: string;
    pos_r, pos_g: integer;
    r, g, b: integer;
    sum : real;
Begin;
    ReadLn (colors);
    RealLn (sign);
    
    pos_r := Pos(colors, ',');
    rs := Copy(colors, 1, pos_r - 1);
    
    pos_g := PosEx(colors, ',', pos_r + 1);
    rg := Copy(colors, pos_r +1, pos_g - 1);
    
    rb := Copy(colors, pos_g + 1, Length(colors));
    
    r := StrToInt(rs);
    g := StrToInt(rg);
    b := StrToInt(rb);
    
    sum := r + g + b;
	
	WriteLn (Contrast);     { 1 }
	WriteLn (r);            { 2 }
	WriteLn (Round(sum / 255); { 3 }	
	WriteLn (Transparanse); { 4 }
	
	If ((r > g) and (r > b) )
		Then begin
			WriteLn (r);
		else begin
			if (g > b) 
				Then begin
					WriteLn (g);
				else begin
					WriteLn (b);
			end;
	end;

	If ((r < g) and (r < b) )
		Then begin
			WriteLn (r);
		else begin
			if (g < b) 
				Then begin
					WriteLn (g);
				else begin
					WriteLn (b);
			end;
	end;
	
	WriteLn (g);            { 7 }
    WriteLn (sign);         { 8 }
end.

{ by Tovarisch Trunaev, 2025, Russia, Uray }
{ Om->Delf(); }
